### PR TITLE
feat: added 'for' & 'uses' into context for relationships

### PR DESCRIPTION
**Pull request by JDFleury**
Adding new variables that will be included in the Authorization entity. 

"for" will be used for the relationship between the Authorization & the gate.
"uses" will be used for the relationship between the Authorization & the key that the authorization is assigned to.

### CODE REVIEW

**Comment by bobeal** [comment]
not really fan of such very generic names.

**Comment by JDFleury** [comment]
> not really fan of such very generic names.

Would you prefer hasGate & hasKey? I have seen this used before within Stellio

File: openIt/ngsild-payloads/authorization.jsonld [hunk 1]
```diff
@@ -13,6 +13,16 @@
       "value": "2024-09-02T11:00:00.000Z"
     }
   },
+  "for": {
+    "type": "Relationship",
+    "object": "urn:ngsi-ld:Gate:123",
+    "datesetId": "urn:ngsi-ld:Dataset:Gate123"
```
[the remark this answers is not in the document]
Comment (JDFleury) [reply openIt/ngsild-payloads/authorization.jsonld:19]: Understood. Will look into the other relationships & properties that are missing as well.

**Comment by bobeal** [comment]
> > not really fan of such very generic names.
> 
> Would you prefer hasGate & hasKey? I have seen this used before within Stellio

in the long term, will open-it open other things that are not gates?

**Comment by JDFleury** [comment]
> > > not really fan of such very generic names.
> > 
> > 
> > Would you prefer hasGate & hasKey? I have seen this used before within Stellio
> 
> in the long term, will open-it open other things that are not gates?

It is possible yes. And we discussed other names for "Gate" but still settled on Gate since in English in ca be used that way as well. Other options I think were AccessPoint & Entrance.

File: openIt/jsonld-contexts/openIt.jsonld [hunk 2]
```diff
@@ -4,13 +4,13 @@
     "Gate": "https://vocab.egm.io/Gate",
     "Key": "https://vocab.egm.io/Key",
     "codes": "https://vocab.egm.io/codes",
-    "for": "https://vocab.egm.io/for",
     "from": "https://vocab.egm.io/from",
+    "hasGate": "https://vocab.egm.io/hasGate",
+    "hasKey": "https://vocab.egm.io/hasKey",
```
Comment (junjou) [inline openIt/jsonld-contexts/openIt.jsonld:9]: In that case what about we prefix with for and uses i.e: `forGate` and `usesKey`

Comment (JDFleury) [reply openIt/jsonld-contexts/openIt.jsonld:9]: I think that works best.

**Review by junjou — APPROVED** [review]

**Pull request merged**